functors_task_1/student/include/impl.h

```diff
@@ -1,13 +1,51 @@
 #pragma once
 #include "bar_serving.h"
 #include <functional>
+#include <iostream>
+#include <map>
+
+using namespace std;
 
 /**
  * @todo Implement functor-generator that will return next beer brand (cyclic)
  * @example HoeGaarden -> Corona -> Carlsberg -> Bud -> ZlataPraha -> Leffe -> HoeGaarden...
  */
+
 struct BeerOrganizer
 {
+    void operator()(){
```

Comment:
That will not work. We need to return BeerBrand, iterating over the enum, casting it to the int